src/backendbot/repositories/ram_repository.py

```
@@ -0,0 +1,101 @@
+            "free_mb": round(mem.free / (1024 * 1024), 2),
+        }
+
+    def get_detailed_ram_metrics_powershell(self) -> Optional[Dict[str, Any]]:
+        """
+        Executes a PowerShell script to get detailed RAM metrics and returns the parsed JSON output.
+        """
+        script_path = "c:/Users/DELL/Desktop/BackendBot/scripts/get_detailed_ram_metrics.ps1"
+        try:
+            # Use powershell.exe for Windows compatibility
+            command = ["powershell.exe", "-ExecutionPolicy", "Bypass", "-File", script_path]
+            result = subprocess.run(command, capture_output=True, text=True, check=True, encoding='utf-8')
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)